pkg/redirectpolicy/manager.go

```diff
@@ -439,6 +450,12 @@ func (rpm *Manager) deletePolicyService(svcID k8s.ServiceID) {
 		config := rpm.policyConfigs[rp]
 		for _, m := range config.frontendMappings {
 			rpm.deletePolicyFrontend(config, m.feAddr)
+			// Retores the svc backends if there's still such a k8s svc.
+			swg := lock.NewStoppableWaitGroup()
+			restored := rpm.svcCache.EnsureService(svcID, swg)
+			if restored {
+				log.Info("Restored service ", svcID)
+			}
```

Comment:
Out of curiosity, instead of deleting and re-creating the service, can't we simply update the service with a new frontend? I'm just thinking out loud here if that's something possible.

Reply:
The restored service has the same frontend with the LRP (it was shadowed by LRP in the first place), not sure how we can use a different frontend?